build/builder.go

```diff
@@ -505,15 +505,44 @@ func (o *Options) FindAllShards() []string {
 
 // IgnoreSizeMax determines whether the max size should be ignored.
 func (o *Options) IgnoreSizeMax(name string) bool {
+	matched := false
 	for _, pattern := range o.LargeFiles {
 		pattern = strings.TrimSpace(pattern)
-		m, _ := doublestar.PathMatch(pattern, name)
+		negated, validatedPattern := checkIsNegatePattern(pattern)
+
+		m, _ := doublestar.PathMatch(validatedPattern, name)
 		if m {
-			return true
+			// match against a negated LargeFiles pattern overrides all non-exclude matches
+			if negated {
+				return false
+			}
+			matched = true
 		}
 	}
 
-	return false
+	return matched
+}
+
+func checkIsNegatePattern(pattern string) (bool, string) {
+	negate := "!"
+
+	if len(pattern) < len(negate) {
+		return false, pattern
+	}
+
+	// if negated then strip prefix meta characters representing negated filter pattern
+	if strings.Index(pattern, negate) == 0 {
```

Comment:
Maybe it is better to use strings.HasPrefix here? I looked at the implementations of both and HasPrefix seems to have less overhead.

Reply:
That makes sense, I'll replace with HasPrefix

Reply:
Indeed this function would be relatively straightforward using HasPrefix. You essentially only need to deal with `strings.HasPrefix(pattern, "!")` and `strings.HasPrefix(pattern, "\\!")`